src/backend/base/langflow/components/vectorstores/astradb.py

```diff
@@ -380,53 +397,62 @@ async def create_collection_api(
-            dimension=dimension,
-            service=vectorize_options,
-        )
+        # Enable the reranker and lexical options if hybrid search is selected
+        rerank_bool = search_method == "Hybrid Search"
+
+        # TODO: ONLY pass rerank and lexical if enviroment is "dev" change this soon
+        base_args = {
+            "collection_name": new_collection_name,
+            "token": token,
+            "api_endpoint": api_endpoint,
+            "keyspace": keyspace,
+            "environment": environment or "prod",
+            "embedding_dimension": dimension,
+            "collection_vector_service_options": vectorize_options,
+        }
+
+        # Add optional arguments only if environment is "dev"
+        if environment == "dev":
+            base_args["collection_rerank"] = CollectionRerankOptions(enabled=rerank_bool)
+            base_args["collection_lexical"] = CollectionLexicalOptions(enabled=rerank_bool)
+
+        _AstraDBCollectionEnvironment(**base_args)
 
     @classmethod
     def get_database_list_static(cls, token: str, environment: str | None = None):
-        client = DataAPIClient(token=token, environment=environment)
+        client = DataAPIClient(environment=environment)
 
         # Get the admin object
         admin_client = client.get_admin(token=token)
 
         # Get the list of databases
-        db_list = list(admin_client.list_databases())
-
-        # Set the environment properly
-        env_string = ""
-        if environment and environment != "prod":
-            env_string = f"-{environment}"
+        db_list = admin_client.list_databases()
 
         # Generate the api endpoint for each database
         db_info_dict = {}
         for db in db_list:
             try:
                 # Get the API endpoint for the database
-                api_endpoint = f"https://{db.info.id}-{db.info.region}.apps.astra{env_string}.datastax.com"
+                api_endpoint = db.regions[0].api_endpoint
 
                 # Get the number of collections
                 try:
+                    # Get the number of collections in the database
                     num_collections = len(
-                        list(
-                            client.get_database(
-                                api_endpoint=api_endpoint, token=token, keyspace=db.info.keyspace
-                            ).list_collection_names(keyspace=db.info.keyspace)
-                        )
+                        client.get_database(
+                            api_endpoint,
+                            token=token,
+                        ).list_collection_names()
                     )
                 except Exception:  # noqa: BLE001
                     if db.status != "PENDING":
                         continue
                     num_collections = 0
 
                 # Add the database to the dictionary
-                db_info_dict[db.info.name] = {
+                db_info_dict[db.name] = {
```

Comment:
Possible (albeit very unlikely) risk: multiple databases with same name. Uniqueness is only on the database ID.
(another minor non-urgent thing)

Reply:
This is an important point. unfortunately the component is kind of set up to assume uniqueness of db names.... i need to think about how to better handle this in the context of langflow, but for now i think i'll open a JIRA and defer this to a later release in hopes that its not going to be hugely impactful to most people. but thanks for calling it out